app/uk/gov/hmrc/agentservicesaccount/controllers/AgentServicesController.scala

```diff
@@ -119,36 +123,57 @@ class AgentServicesController @Inject()(
 
   val manageAccount: Action[AnyContent] = Action.async { implicit request =>
     withAuthorisedAsAgent { agentInfo =>
-        if (agentInfo.isAdmin) {
-          if (appConfig.granPermsEnabled) {
-            agentPermissionsConnector.isArnAllowed flatMap {
-              case true =>
-                for {
-                  _ <- agentPermissionsConnector.syncEacd(agentInfo.arn)
-                  status <- agentPermissionsConnector.getOptinStatus(agentInfo.arn)
-                  mGroups <- agentPermissionsConnector.getGroupsSummaries(agentInfo.arn)
-                  hasAnyGroups = mGroups.exists(_.groups.nonEmpty)
-                } yield {
-                  Ok(manageAccountView(status.map(ManageAccessPermissionsConfig(_, hasAnyGroups))))
-                }
-              case false =>
-                Future successful Ok(manageAccountView(None))
-            }
-          } else {
-            Future.successful(Ok(manageAccountView(None)))
+      if (agentInfo.isAdmin) {
+        if (appConfig.granPermsEnabled) {
+          agentPermissionsConnector.isArnAllowed flatMap {
+            case true =>
+              for {
+                _ <- agentPermissionsConnector.syncEacd(agentInfo.arn)
+                status <- agentPermissionsConnector.getOptinStatus(agentInfo.arn)
+                mGroups <- agentPermissionsConnector.getGroupsSummaries(agentInfo.arn)
+                hasAnyGroups = mGroups.exists(_.groups.nonEmpty)
+              } yield {
+                Ok(manage_account(status.map(ManageAccessPermissionsConfig(_, hasAnyGroups))))
+              }
+            case false =>
+              Future successful Ok(manage_account(None))
           }
         } else {
-          Future.successful(Forbidden)
+          Future.successful(Ok(manage_account(None)))
         }
+      } else {
+        Future.successful(Forbidden)
+      }
     }
   }
 
+  val yourAccount: Action[AnyContent] = Action.async { implicit request =>
+    withFullUserDetails { (agentInfo: AgentInfo) =>
+      if (!agentInfo.isAdmin) {
+        if (appConfig.granPermsEnabled) {
+          agentInfo.credentials.fold(Ok(your_account(None)).toFuture) { credentials =>
```

Comment:
tbh i just copied the manage-account code which is the same. So i guess it's ok, otherwise we need to change both. maybe a redirect to home might be more appropriate in both cases?